### PR TITLE
Hide client secrets in repl

### PR DESCRIPTION
**Pull request by NRHelmi**
This PR hides client secrets in `Content` and `Config` when printing in repl:
```
julia> ctx
Context("us-east", "https", "azure.relationalai.com", "443", (y8Ftwa6KV6Tnvpd4APTtisiojUHIcUvu, m7d..., (eyJ..., run:transaction read:transaction update:database list:database delete:database create:database list:compute read:compute delete:compute create:compute list:user create:accesskey list:accesskey create:oauth_client update:oauth_client list:oauth_client delete:oauth_client read:oauth_client list:permission read:role list:role delete:user create:user read:user update:user read:credits_usage rotate:oauth_client_secret list:transaction cancel:transaction, 3600, 2022-08-12T17:49:51.365), https://login.relationalai.com/oauth/token), "https://azure.relationalai.com")

julia> conf
Config("us-east", nothing, "azure.relationalai.com", "443", (y8Ftwa6KV6Tnvpd4APTtisiojUHIcUvu, m7d..., (eyJ..., run:transaction read:transaction update:database list:database delete:database create:database list:compute read:compute delete:compute create:compute list:user create:accesskey list:accesskey create:oauth_client update:oauth_client list:oauth_client delete:oauth_client read:oauth_client list:permission read:role list:role delete:user create:user read:user update:user read:credits_usage rotate:oauth_client_secret list:transaction cancel:transaction, 3600, 2022-08-12T17:49:51.365), https://login.relationalai.com/oauth/token), nothing)

julia> conf.credentials.access_token
(eyJ..., run:transaction read:transaction update:database list:database delete:database create:database list:compute read:compute delete:compute create:compute list:user create:accesskey list:accesskey create:oauth_client update:oauth_client list:oauth_client delete:oauth_client read:oauth_client list:permission read:role list:role delete:user create:user read:user update:user read:credits_usage rotate:oauth_client_secret list:transaction cancel:transaction, 3600, 2022-08-12T17:49:51.365)

### CODE REVIEW

File: src/creds.jl [hunk 1]
```diff
@@ -36,3 +48,16 @@ mutable struct ClientCredentials <: Credentials
...
         new(client_id, client_secret, client_credentials_url, nothing)
 end
+
+function Base.show(io::IO, c::ClientCredentials)
+    print(
+        io,
+        "(",
+        c.client_id,
+        c.client_secret == nothing ? "" : ", $(c.client_secret[1:3])...",
+        c.access_token == nothing ? "" : ", $(c.access_token)",
```
Comment (larf311) [inline src/creds.jl:58]: Shouldn't this also have the [1:3]?

Comment (NRHelmi) [reply src/creds.jl:58]: `access_token` is a struct and it's handled in a different show function `function Base.show(io::IO, t::AccessToken)`

**Comment by larf311** [comment]
Are there any unit tests? ;)

File: test/api.jl [hunk 2]
```diff
@@ -202,3 +203,13 @@ end
...
+
+@testset "hide client secrets in repl" begin
+    access_token = AccessToken("abc_token", "run:transaction", 3600, DateTime("2022-08-12T17:49:51.365"))
+    creds = ClientCredentials("client_id", "xyz_client_secret", "https://login.relationalai.com/oauth/token")
+    creds.access_token = access_token
+
+    io = IOBuffer()
+    show(io, creds)
+    @test String(take!(io)) === "(client_id, xyz..., (abc..., run:transaction, 3600, 2022-08-12T17:49:51.365), https://login.relationalai.com/oauth/token)"
+end
```
Comment (NHDaly) [inline test/api.jl:215]: missing endline
(There's a setting to enable always adding an endline in VSCode if you use it, which is nice):
unnamed

File: src/creds.jl [hunk 3]
```diff
@@ -36,3 +48,16 @@ mutable struct ClientCredentials <: Credentials
...
+        ", ",
+        c.client_credentials_url,
+        ")"
+    )
+end
```
Comment (NHDaly) [inline src/creds.jl:63]: 🤔 shouldn't you not need this? I think that it should automatically default to recursively calling show() on the inner arguments. For example:
```julia
julia> struct A
           x::Int
       end

julia> Base.show(io::IO, a::A) = print(io, "HELLO WORLD: $(a.x)")

julia> struct B
           a::A
           y::Int
       end

julia> B(A(1), 2)
B(HELLO WORLD: 1, 2)
```

Comment (NHDaly) [reply src/creds.jl:63]: oh, i'm sorry never mind. I see now that you're also trying to hide the client_secret in this struct. 👍

File: src/creds.jl [hunk 4]
```diff
@@ -36,3 +48,16 @@ mutable struct ClientCredentials <: Credentials
     ClientCredentials(client_id, client_secret, client_credentials_url = nothing) =
         new(client_id, client_secret, client_credentials_url, nothing)
 end
```
Comment (NHDaly) [inline src/creds.jl:50]: Can we add a note to the docstring for this struct explaining that the client_secret is hidden by default when displaying this object, and guiding them towards how to actually print the secret if they want to?

They could always do `ctx.credentials.client_secret`, and that will show it, but maybe they wouldn't know how to do that? 🤔 

I dunno, i can't shake the feeling that someone might actually _want_ to print their credentials, and won't be able to figure out how to.

**Review by NHDaly — APPROVED** [review]
LGTM thanks helmi!

**Pull request merged**